hack/internal/tools/yaml-processor/main.go

```diff
@@ -32,7 +34,10 @@ func main() {
 		log.Fatalf("Failed to initialize logger: %v", err)
 	}
 	defer func() {
-		if err := logger.Sync(); err != nil {
+		// Sync logger and handle errors, ignoring specific non-critical errors for console outputs.
+		// Ignores syscall.ENOTTY (MacOS: "inappropriate ioctl for device") and syscall.EINVAL (Linux: "invalid argument")
+		// when syncing to non-file descriptors like /dev/stderr or /dev/stdout.
+		if err := logger.Sync(); err != nil && !errors.Is(err, syscall.ENOTTY) && !errors.Is(err, syscall.EINVAL) {
```

Comment:
Why the "invalid argument" error is something we would like to silence as non-critical? Can you provide example error like this?

Reply:
hm, any idea why they are logged in the first place? I'm wondering if we could swipe a legitimate error like match the regex. the regex does not feel particularly specific.

Reply:
I can also add an additional check to ensure the error has the prefix sync /dev/stderr for more specificity. It looks like both errors include this prefix.